FindLibKrb5.cmake

```
@@ -18,13 +18,26 @@
 #  LibKrb5_INCLUDE_DIR             The location of Krb5 headers
 
 if (NOT LibKrb5_ROOT_DIR)
-    find_path(LibKrb5_ROOT_DIR NAMES include/krb5.h)
+    find_path(LibKrb5_ROOT_DIR NAMES include/krb5/krb5.h)
 endif ()
 
 find_library(LibKrb5_LIBRARY NAMES krb5 HINTS ${LibKrb5_ROOT_DIR}/lib)
 
 find_path(LibKrb5_INCLUDE_DIR NAMES krb5/krb5.h HINTS ${LibKrb5_ROOT_DIR}/include)
 
+# If using macOS, make sure that we're not finding the system's libkrb5. It
+# an old version, which points you at GSS.framework, which is also deprecated.
+if (LibKrb5_LIBRARY AND "${CMAKE_SYSTEM_NAME}" MATCHES "Darwin")
+    string(FIND "${LibKrb5_LIBRARY}" "/Library/Developer/CommandLineTools/SDKs"
```

Nit, but also didn't test.

```suggestion
    if ( "${LibKrb5_LIBRARY}" MATCHES "..." )
        message(FATAL_ERROR)
    endif ()
```

I'm also wondering if there's a define or something in `krb5.h` that could be used for identifying the wrong/too-old version, but if what you have works :+1:

---

> I'm also wondering if there's a define or something in krb5.h that could be used for identifying the wrong/too-old version, but if what you have works

I looked through it and couldn't find a version number or anything unforunately. The best indicator I could find of it being very old was simply the copyright information. I couldn't tell you what version Apple is actually including here without pulling all of the old tarballs and diffing. It's at least 12 years old, because https://github.com/krb5/krb5/commit/17e205070745e4712d40eed32d92d02edb47009f changed everything to use `stdint.h` types and Apple's version doesn't include that.

---

> Nit, but also didn't test.

`string(FIND)` is probably better here because the path changes as you install new versions of the macOS SDK:

```
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/lib/libkrb5.tbd
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/lib/libkrb5.tbd
/Library/Developer/CommandLineTools/SDKs/MacOSX14.5.sdk/usr/lib/libkrb5.tbd
/Library/Developer/CommandLineTools/SDKs/MacOSX15.4.sdk/usr/lib/libkrb5.tbd
```

I suppose it could match a regex. We definitely do `if MATCHES` more often than we do `string(FIND)`.